allennlp/data/dataset_readers/wikitables/wikitables_dataset_reader.py

```diff
@@ -258,33 +283,23 @@ def text_to_instance(self,  # type: ignore
         return Instance(fields)
 
     @staticmethod
-    def _parse_line_as_lisp(lisp_string: str) -> Dict[str, Union[str, List[str], None]]:
+    def _parse_example_line(lisp_string: str) -> Dict[str, Union[str, List[str], None]]:
         """
         Training data in WikitableQuestions comes with examples in the form of lisp strings in the format:
             (example (id <example-id>)
                      (utterance <question>)
                      (context (graph tables.TableKnowledgeGraph <table-filename>))
                      (targetValue (list (description <answer1>) (description <answer2>) ...)))
 
-        We parse such strings and return the parsed information here.
+        We parse such strings and return the parsed information here.  We don't actually use the
+        target value right now, because we use a pre-computed set of logical forms.  So we don't
+        bother parsing it; we can change that if we ever need to.
         """
-        parsed_info = {}
-        input_nested_list = pyparsing.OneOrMore(pyparsing.nestedExpr()).parseString(lisp_string).asList()[0]
-        # Skipping "example"
-        for key, value in input_nested_list[1:]:
-            if key == "id":
-                parsed_info["id"] = value
-            elif key == "utterance":
-                parsed_info["question"] = value.replace("\"", "")
-            elif key == "context":
-                # Skipping "graph" and "tables.TableKnowledgeGraph"
-                parsed_info["context"] = value[-1]
-            elif key == "targetValue":
-                # Skipping "list", and "description" within each nested list.
-                parsed_info["targetValue"] = [x[1] for x in value[1:]]
-        # targetValue may not be present if the answer is not provided.
-        assert all([x in parsed_info for x in ["id", "question", "context"]]), "Invalid format"
-        return parsed_info
+        id_piece, rest = lisp_string.split(') (utterance "')
```

Comment:
You're assuming the fields in the string come in a specific order. It may be a good idea to make sure the order of these fields is what you expect it to be (i.e. check the indices of the field names).
Or you could use regex. Is that significantly slower?
Not really a serious issue though. The only situation this would be a problem is if at some point of time we switch to using a different version of the dataset that is formatted differently.

Reply:
The code will crash if these fields come in a different order, because of the way I'm assigning to a tuple after the calls to `.split()`.  I think that's good enough for now, and if there ever comes a time when we switch to some different version of the dataset, we can worry about updating this then.